exam/exam.py

```diff
@@ -198,19 +199,20 @@ def _compute_per(self):
     @api.depends('result_ids','student_id')
     def _compute_result(self):
         flag = False
-        if self.result_ids and self.student_id:
-            for l in self.result_ids:
-                if self.student_id.year.grade_id.grade_ids:
-                    for grades in self.student_id.year.grade_id.grade_ids:
-                        if grades.grade:
-                            if not grades.fail:
-                                self.result = 'Pass'
-                            else:
-                                flag=True
-                else:
-                    raise except_orm(_('Configuration Error !'), _('First Select Grade System in Student->year->.'))
-        if flag:
-            self.result = 'Fail'
+        for result_obj in self:
+            if result_obj.result_ids and result_obj.student_id:
+                for l in result_obj.result_ids:
+                    if result_obj.student_id.year.grade_id.grade_ids:
```

Comment:
Do not use variable 'l'! Please add comments to know what you are trying to achieve!
Please check whether we can reduce the complexity, the 3 nested loops are way too much!